crates/red_knot_python_semantic/src/semantic_index/definition.rs

```diff
@@ -9,6 +9,19 @@ use crate::semantic_index::symbol::{FileScopeId, ScopeId, ScopedSymbolId};
 use crate::unpack::Unpack;
 use crate::Db;
 
+/// A definition of a symbol.
+///
+/// ## Module local type
+/// This type should not be used as part of any cross module API because
+/// it holds a reference to the AST node and range offset changes
+/// then propagate through all usages and deserialization requires
+/// reparsing the entire module.
```

Comment:
```suggestion
/// ## Module-local type
/// This type should not be used as part of any cross-module API because
/// it holds a reference to the AST node. Range-offset changes
/// then propagate through all usages, and deserialization requires
/// reparsing the entire module.
```